tests/test_virtual.py

```diff
@@ -137,6 +142,14 @@ def draw_fn(draw, width, height):
 
     virtual.add_hotspot(widget, (19, 56))
     virtual.set_position((28, 30))
+    virtual.remove_hotspot(widget, (19, 56))
 
     bbox = ImageChops.difference(reference, device.image).getbbox()
     assert bbox is None
+
+
+def test_terminal_println():
+    device = dummy()
+    ter = terminal(device)
+
+    ter.println('foo\rbar\bspam\teggs\n')
```

Comment:
I think the best you can hope for here is a regression test - much like the others in this file:

* load a reference image
* do the operation
* diff the resulting image with the reference image 
* assert that the bounding box is None

In order to generate the reference image, I would just do a one-off `(device.image.save('tests/reference/terminal.py')` and visually check that the resultant image looks right.

Also, this technique can be used to check that the terminal lines up tabs, eg.

```python
ter.println("1\t32\t999\n")
ter.println("999\t1\t32\n")
```

And double newlines should have a blank line between text,

```python
ter.println("line1\n\nline2\n")
```

Lastly, check that wrapping to the device width works as expected